handlers/messagebird/messagebird.go

```diff
@@ -0,0 +1,240 @@
+)
+
+func init() {
+	courier.RegisterHandler(newHandler("MBD", "Messagebird", true))
+}
+
+type handler struct {
+	handlers.BaseHandler
+	validateSignatures bool
+}
+
+func newHandler(channelType courier.ChannelType, name string, validateSignatures bool) courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("MBD"), "Messagebird"), validateSignatures}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	s.AddHandlerRoute(h, http.MethodPost, "receive", courier.ChannelLogTypeMsgReceive, handlers.JSONPayload(h, h.receiveMessage))
+	return nil
+}
+
+func (h *handler) receiveMessage(ctx context.Context, channel courier.Channel, w http.ResponseWriter, r *http.Request, payload *ReceivedMessage, clog *courier.ChannelLog) ([]courier.Event, error) {
+	err := h.validateSignature(channel, r)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	err = handlers.DecodeAndValidateJSON(payload, r)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	// no message? ignore this
+	if payload.Body == "" && !payload.Mms {
+		return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "Ignoring request, no message")
+	}
+
+	// create our date from the timestamp
+	date := payload.CreatedDatetime
+
+	// create our URN
+	urn, err := urns.NewTelURNForCountry(payload.Originator, "US")
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+	text := payload.Body
+
+	// build our msg
+	msg := h.Backend().NewIncomingMsg(channel, urn, text, clog).WithReceivedOn(date.UTC()).WithExternalID(payload.ID)
+
+	// process any attached media
+	if payload.Mms {
+		for i := 0; i < len(payload.MediaURLs); i++ {
+			msg.WithAttachment(payload.MediaURLs[i])
+		}
+	}
+	// and finally write our message
+	return handlers.WriteMsgsAndResponse(ctx, h, []courier.Msg{msg}, w, r, clog)
+}
+
+func (h *handler) Send(ctx context.Context, msg courier.Msg, clog *courier.ChannelLog) (courier.MsgStatus, error) {
+
+	authToken := msg.Channel().StringConfigForKey(courier.ConfigAuthToken, "")
+	if authToken == "" {
+		return nil, fmt.Errorf("missing config 'auth_token' for Messagebird channel")
+	}
+
+	user := msg.URN().Path()
+	status := h.Backend().NewMsgStatusForID(msg.Channel(), msg.ID(), courier.MsgErrored, clog)
+
+	// create base payload
+	payload := &Message{
+		Recipients: []string{user},
+		Originator: msg.Channel().Address(),
+	}
+	// build message payload
+
+	if len(msg.Text()) > 0 {
+		payload.Body = msg.Text()
+	}
+	sendUrl := ""
+	if len(msg.Attachments()) > 0 {
+		sendUrl = mmsURL
+	} else {
+		sendUrl = smsURL
+	}
+	for _, attachment := range msg.Attachments() {
+		mediaType, mediaURL := handlers.SplitAttachment(attachment)
+		switch strings.Split(mediaType, "/")[0] {
+		case "image":
```

Comment:
does messagebird support other attachments such as video, audio or docs files?
Can those be added as well?

Reply:
yup. done